pkg/utils/events.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+// For each chaos resource, the controller manager will record events
+// once the chaos is started or completed or failed. The starting and
+// completing events should be of type "Normal". And the failed events should
+// be of type "Warning". The reasons are defined as following.
+const (
+	// The chaos just started
+	EventChaosToInject string = "ChaosToInject"
+
+	// The chaos just failed when injecting. The message should include detailed error
+	EventChaosInjectFailed string = "ChaosInjectFailed"
+
+	// The chaos just failed when recovering. The message should include detailed error
+	EventChaosRecoverFailed string = "ChaosRecoverFailed"
```

Comment:
I think these types could be merged into one event type `Reconciliation failed type`.

Reply:
I think we should throw detailed error as soon as we could. Just as we discussed in #207 . What do you think? @zhouqiang-cl  @Yisaer

Reply:
I think two failed is OK. if not using two types, we should also in somewhere to tell what errors

Reply:
IMO, if we want to expose the detailed error message for event chaos inject or event chaos recover, I think we should send the event in the chaos action `types.go` like `EventChaosSuccess` instead of send it in the end of the reconciliation.

Reply:
If we put specific error events inside `types.go`, then we may miss errors during scheduling. And there are lots of error branches inside it, it's boring to record each of them repeatedly. I think we can do a little bit better via seperating injection failures and recovery failures for now.

Reply:
We could refine the error recording procedure in future PRs for getting more friendly errors.

Reply:
Yes, I recommended discussing the error event in another issue. For now, I think the event success issue is enough.

Reply:
OK. I will resolve this conversation.